src/rpc/core_rpc_server.cpp

```diff
@@ -74,7 +74,7 @@ namespace
   void store_difficulty(cryptonote::difficulty_type difficulty, uint64_t &sdiff, std::string &swdiff, uint64_t &stop64)
   {
     sdiff = (difficulty << 64 >> 64).convert_to<uint64_t>();
-    swdiff = difficulty.convert_to<std::string>();
+    swdiff = cryptonote::hex(difficulty);
```

Comment:
This type has a built-in function `difficulty.str(0, std::ios_base::hex);` which looks to have the same output, except the characters are upper case. The leading "0x" is also output. It does not use a `std::stringstream` internally, and the implementation looks to be similar in performance.